.github/workflows/build-full-application.yaml

```diff
@@ -0,0 +1,99 @@
+name: Build Full Application workflow
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: "The name of the environment to build into"
+        required: true
+        type: string
+      workspace:
+        description: "The name of the workspace to build the application into"
+        required: true
+        type: string
+      services:
+        description: "A list of the services to build"
+        required: false
+        default: ""
+        type: string
+      tag:
+        description: "The tag to build from"
+        required: false
+        default: ""
+        type: string
+      multi_domain:
+        description: "Domain to use for github runner role"
+        required: false
+        default: ""
+        type: string
+      workflow_timeout:
+        description: "Timeout duration in minutes"
+        required: false
+        default: 5
+        type: number
+      target_repo:
+        description: "The name of the repo"
+        required: true
+        type: string
+      target_repo_commit_hash:
+        description: "The name of the target repo branch"
+        required: true
+        type: string
+
+jobs:
+  deploy-infrastructure-apply:
+    name: "Apply Deploy Infrastructure"
+    strategy:
+      matrix:
+        domain: ['uec-dos-service-management','uec-cm']
+    uses: ./.github/workflows/deploy-infrastructure.yaml
+    with:
+      environment: ${{ inputs.environment }}
+      workspace: ${{ inputs.workspace }}
+      domain: ${{ matrix.domain }}
+      multi_domain: ${{ inputs.multi_domain }}
+      stacks: "['application']"
```

Comment:
Again maybe for later but we need to deploy data-migration for service management - there may be other domain-specific stacks that need to be deployed too later. For this we may need a separate job per stack

Reply:
We will have to think about this quite deeply, as deploying different stacks in different domains makes things extremely challenging. Each of these operations are performed within a matrix step, and then each matrix step spawns another series of matrix steps (based on the services within that directory).

If we did have an additional job for 'data-migration' - then this job would fail when the matrix step is executed for a domain which does not have a 'data-migration' stack. We would need to find a way to allow it to fail intelligently without closing the pipeline.

Reply:
ticket to be created

Reply:
DR-1027 - suggesting adding domain-specific job to handle non-standard stacks - avoiding the use of matrices for domain (but potentially adding it for stacks)